qiskit/transpiler/passes/basis/basis_translator.py

```diff
@@ -90,6 +90,12 @@ def __init__(self, equivalence_library, target_basis, target=None):
 
         super().__init__()
         self._equiv_lib = equivalence_library
+        if target_basis and "u" in target_basis:
+            DeprecationWarning(
+                'Using "u" in the basis for referring to the generic single-qubit rotation gate is '
+                'being deprecated in favor of "U". Please use capital "U" in the future.'
```

Comment:
It would be good to add the release version (also for us, so we know when we can drop support 😉)
```suggestion
                'Using "u" in the basis for referring to the generic single-qubit rotation gate is '
                'being deprecated as of Qiskit Terra 0.20.0 and will not be supported anymore '
                'at earliest 3 months after the release date. Please use capital "U" in the future.'
```